metric-schema-java/src/main/java/com/palantir/metric/schema/UtilityGenerator.java

```diff
@@ -506,13 +519,11 @@ private static void generateMetricFactoryBuilder(
                 .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
                 .returns(MetricName.class)
                 .build();
-        List<MethodSpec> abstractBuildMethods = isGauge
-                ? ImmutableList.of(abstractBuildMethod, abstractBuildMetricName)
-                : ImmutableList.of(abstractBuildMethod);
 
         outerBuilder.addType(TypeSpec.interfaceBuilder(buildStage(metricName))
                 .addModifiers(visibility.apply())
-                .addMethods(abstractBuildMethods)
+                .addMethod(abstractBuildMethod)
+                .addMethod(abstractBuildMetricName)
```

Comment:
Looks like an oversight in the initial implementation, we should add a `@CheckReturnValue` annotation to the `buildMetricName` method.